snowfakery/parse_recipe_yaml.py

```diff
@@ -375,6 +385,23 @@ def parse_object_template(yaml_sobj: Dict, context: ParseContext) -> ObjectTempl
 
         if count_expr is not None:
             parse_count_expression(yaml_sobj, sobj_def, context)
+
+        for_each_expr = yaml_sobj.get("for_each")
+        if for_each_expr is not None:
+            # it would be easy to support multiple parallel
+            # for-eaches here and at one point the code did,
+            # but the use-case wasn't obvious so it was removed
+            # after 9bc296a7df. If we get to 2023 without
+            # finding a use-case we can delete this comment.
+            if isinstance(for_each_expr, dict):
+                sobj_def["for_each_expr"] = parse_for_each_variable_definition(
+                    for_each_expr, context
+                )
+            else:  # pragma: no cover
```

Comment:
Why the pragma here? This seems like something that ought to be tested.

Reply:
The pragma is because this is sort of a "backup check". The code is unreachable if a test further up in the call chain did its job properly. I'll add a comment.

Reply:
Added comment:

https://github.com/SFDO-Tooling/Snowfakery/pull/457/commits/b7054d2b2adf278da6ba5fbcc1be10ebcb9aa76a#diff-57b088a5fa3b0961b21872462d9e327874cb6ce9b5c7f490a05c3f5b30c33b18R401